src/libcore/slice/mod.rs

```diff
@@ -400,14 +400,14 @@ impl<T> SliceExt for [T] {
         while size > 1 {
             let half = size / 2;
             let mid = base + half;
-            // mid is always in [0, size).
+            // mid is always in [0, size].
```

Comment:
Maybe a connective here, e.g., "mid is always in [0, size), i.e., [bullet points below]"